File: yh_sum/scripts/yh_sum_sub.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

import rospy
from yh_sum.msg import yh_sum_msg

sum = 0
def msgCallback(msg):
    rospy.loginfo("receieve message: %d", msg.stamp.secs)
    rospy.loginfo("receieve message: %d", msg.stamp.nsecs)
    rospy.loginfo("receieve message: %d", msg.data)
    global sum
    sum = sum + msg.data
    print(sum)


    

def listner():
    rospy.init_node("yh_sum_sub", anonymous=True)
    rospy.Subscriber("yh_sum_topic", yh_sum_msg, msgCallback)

    rospy.spin()

if __name__ == "__main__":
    listner()

File: yh_sum/src/yh_sum_sub.cpp
#include "ros/ros.h"
#include "yh_sum/yh_sum_msg.h"

int sum = 0;

class MyClass
{
    public:
        MyClass()
        {
            sub = nh.subscribe("yh_sum_topic", 100, &MyClass::msgCallback, this);
        }

        void msgCallback(const yh_sum::yh_sum_msg::ConstPtr& msg); 


        // void msgCallback(const yh_sum::yh_sum_msg::ConstPtr& msg)
        // {
        //     ROS_INFO("receive msg = %d", msg->stamp.sec);
        //     ROS_INFO("receive msg = %d", msg->stamp.nsec);
        //     ROS_INFO("receive msg = %d", msg->data);
        //     sum = sum + msg->data;
        //     printf("정수의 합은 sum : %d \n", sum);
        // }

    private:
        ros::NodeHandle nh;
        ros::Subscriber sub;
        int sum = 0;

};

void MyClass::msgCallback(const yh_sum::yh_sum_msg::ConstPtr& msg)
{
    ROS_INFO("receive msg = %d", msg->stamp.sec);
    ROS_INFO("receive msg = %d", msg->stamp.nsec);
    ROS_INFO("receive msg = %d", msg->data);
    sum = sum + msg->data;
    printf("정수의 합은 sum : %d \n", sum);
}

int main(int argc, char** argv)
{
    ros::init(argc, argv, "yh_sum_sub");
    
    MyClass my_class;
    // ros::NodeHandle nh;
    // ros::Subscriber sub =  nh.subscribe("yh_sum_topic", 100, msgCallback);
    ros::spin();  //프로그램 종료 방지 대기
    return 0;
}


